src/DssDeploy.t.sol

```diff
@@ -544,12 +552,24 @@ contract DssDeployTest is DssDeployTestBase {
         assertEq(gov.balanceOf(address(user1)), 1 ether);
     }
 
-    function testFireESM() public {
+    function testFireESMBug() public {
+        deploy();
+        gov.mint(address(user1), 10);
+
+        user1.doESMJoin(address(gov), address(esmBug), 10);
+        assertEq(vat.wards(address(pause.proxy())), 1);
+        esmBug.fire();
+        assertEq(vat.wards(address(pause.proxy())), 1);
+    }
+
+    function testFireESMAttack() public {
         deploy();
         gov.mint(address(user1), 10);
 
-        user1.doESMJoin(address(gov), address(esm), 10);
-        esm.fire();
+        user1.doESMJoin(address(gov), address(esmAttack), 10);
+        assertEq(vat.wards(address(pause.proxy())), 1);
+        esmAttack.fire();
+        assertEq(vat.wards(address(pause.proxy())), 0);
```

Comment:
We might also want to add a `assertEq(end.live(), 0);`

An even more robust test would be to call `esmAttack.deny(clip)` on the ESM and then assert the deny happened on the `clip`.

Reply:
Done